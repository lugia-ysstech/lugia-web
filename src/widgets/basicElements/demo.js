import React, { Component } from 'react';
import BasicElements from './';
import Widgets from '../consts/index';
import Theme from '../theme/index';
import { getBorderRadius, getBorder, getBoxShadow } from '@lugia/theme-utils';
class Demo extends Component {
  render() {
    return (
      <div>
        <Theme
          config={{
            [Widgets.BasicElements]: {
              Container: {
                normal: {
                  width: 200,
                  height: 300,
                  background: {
                    color: 'green',
                  },
                  borderRadius: getBorderRadius(20),
                  border: getBorder({ color: 'blue', width: 1, style: 'dashed' }),
                  boxShadow: getBoxShadow('2 2 20 red'),
                  margin: { top: 20, right: 20, bottom: 20, left: 20 },
                  opacity: 0.6,
                },
              },
            },
          }}
        >
          <BasicElements />
          <BasicElements shape={'circular'} />
          <BasicElements shape={'triangle'} />
        </Theme>
      </div>
    );
  }
}

export default Demo;
