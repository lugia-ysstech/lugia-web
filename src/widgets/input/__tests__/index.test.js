//@flow
import React from 'react';
import Input from '../';
import Theme from '../../theme';
import renderer from 'react-test-renderer';
import chai from 'chai';
import 'jest-styled-components';

import {
  assertInputValue,
  testFireNullKeyBoardEvent,
  testKeyBoardEvent,
  testPropsValue,
} from './InputTestUtils';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Widget from '../../consts';
import { TopInput, ValidateInput } from '../demo';
import type { ValidateType, ValidateStatus } from '../../css/validateHoc';

Enzyme.configure({ adapter: new Adapter() });

const { expect: exp } = chai;
const { mockFunction, VerifyOrder, VerifyOrderConfig } = require('@lugia/jverify');

class LimitInputBox extends React.Component<any, any> {
  static getDerivedStateFromProps(nextProps: Object, preState: Object) {
    if (!preState) {
      return { value: nextProps.value };
    }
    return {};
  }

  onChange = ({ newValue: value }: any) => {
    this.setState({ value });
  };

  render() {
    return <Input value={this.state.value} onChange={this.onChange} />;
  }
}

describe('Input', () => {
  let order;
  let mockFunc;
  let onChange;
  beforeEach(() => {
    order = VerifyOrder.create();
    mockFunc = mockFunction.create(VerifyOrderConfig.create('onChange', order));
    onChange = mockFunc.getFunction();
  });

  function testOnChange(
    title: string,
    props: Object,
    keyPressValue: Array<any>,
    stateValue: Array<any>,
    changeEventValue: Array<any>
  ) {
    it(`props: onchange ${title} `, () => {
      const order = VerifyOrder.create();
      mockFunc = mockFunction.create(VerifyOrderConfig.create('onChange', order));
      const onChange = mockFunc.getFunction();

      const component = mount(<Input {...props} onChange={onChange} />);
      keyPressValue &&
        keyPressValue.forEach((value: any, index: number) => {
          component.find('input').simulate('change', { target: { value } });
          assertInputValue(component, stateValue[index]);
        });

      order.verify(({ onChange }) => {
        changeEventValue &&
          changeEventValue.forEach((v, i) => {
            const { value } = props;
            onChange({
              newValue: v,
              oldValue: stateValue[i - 1] ? stateValue[i - 1] : value ? value : '',
              event: VerifyOrder.Any,
            });
          });
      });
    });
  }

  it('props: null', () => {
    expect(renderer.create(<Input />).toJSON()).toMatchSnapshot();
  });

  function sizeTest(size) {
    it('props: size', () => {
      expect(renderer.create(<Input size={size} />).toJSON()).toMatchSnapshot();
    });
  }

  sizeTest('small');
  sizeTest('large');

  function placeholderTest(placeholder) {
    it('props: placeholder ', () => {
      const jsx = <Input placeholder={placeholder} />;
      const component = mount(jsx);
      expect(renderer.create(jsx).toJSON()).toMatchSnapshot();
      exp(component.props().placeholder).to.be.equal(placeholder);
    });
  }

  placeholderTest('请填写内容');
  placeholderTest('');

  it('props: getTheme: width:100,margin:10,', () => {
    const view = {
      [Widget.Input]: {
        Container: {
          normal: {
            width: 100,
            margin: 10,
          },
        },
      },
    };
    const jsx = (
      <Theme config={view}>
        <Input />
      </Theme>
    );
    expect(renderer.create(jsx).toJSON()).toMatchSnapshot();
  });

  it('props:disabled  onchange', async () => {
    let res = 0;
    const onChange = () => {
      res++;
    };
    const target = mount(
      <Input disabled onChange={onChange}>
        hello
      </Input>
    );
    const input = target.find('input').at(0);
    input.simulate('change');
    expect(res).toBe(0);
    input.simulate('change');
    expect(res).toBe(0);
  });

  it('props: prefix', () => {
    const text = 'hello ligx';
    const prefix = <div className="prefix">{text}</div>;
    const component = mount(<Input prefix={prefix} />);
    exp(component.find('.prefix').text()).to.be.equal(text);
  });

  it('props: suffix', () => {
    const text = 'hello suffix';
    const suffix = <div className="suffix">{text}</div>;
    const component = mount(<Input suffix={suffix} />);
    exp(component.find('.suffix').text()).to.be.equal(text);
  });

  it('props: prefix&suffix', () => {
    const prefixTxt = 'hello ligx';
    const suffixTxt = 'hello kxy';
    const prefix = <div className="prefix">{prefixTxt}</div>;
    const suffix = <div className="suffix">{suffixTxt}</div>;

    const component = mount(<Input prefix={prefix} suffix={suffix} />);
    exp(component.find('.prefix').text()).to.be.equal(prefixTxt);
    exp(component.find('.suffix').text()).to.be.equal(suffixTxt);
  });

  it('props: value', () => {
    const value = '诸法为空';
    testPropsValue(value, value);
  });

  it('props: keyboard event is null but fire keyup event', () => {
    testFireNullKeyBoardEvent('onKeyUp', { keyCode: 49, Target: Input });
    testFireNullKeyBoardEvent('onKeyDown', { keyCode: 49, Target: Input });
    testFireNullKeyBoardEvent('onKeyPress', { keyCode: 49, Target: Input });
    testFireNullKeyBoardEvent('onFocus', { keyCode: 49, Target: Input });
  });
  it('props: onKeyUp', () => {
    testKeyBoardEvent(order, 'onKeyUp', { keyCode: 49, Target: Input });
  });
  it('props: onKeyPress', () => {
    testKeyBoardEvent(order, 'onKeyPress', { keyCode: 49, Target: Input });
  });
  it('props: onKeyDown', () => {
    testKeyBoardEvent(order, 'onKeyDown', { keyCode: 49, Target: Input });
  });

  it('props: onFocus', () => {
    testKeyBoardEvent(order, 'onFocus', { keyCode: 49, Target: Input });
  });
  it('props: onBlur', () => {
    testKeyBoardEvent(order, 'onBlur', { keyCode: 49, Target: Input });
  });
  it('props: onEnter Fired', () => {
    const mockFunc = mockFunction.create(VerifyOrderConfig.create('onEnter', order));

    const keyCode = 13;
    const event = { keyCode };
    mockFunc.mock(({ keyCode }) => {
      exp(keyCode).to.be.equal(keyCode);
    });
    const props = {
      onEnter: mockFunc.getFunction(),
    };

    const component = mount(<Input {...props} />);
    component.find('input').simulate('onKeyDown'.substr(2).toLowerCase(), event);
    order.verify(arg => {
      arg.onEnter(VerifyOrder.Object);
    });
  });

  it('props: onEnter Not Fired', () => {
    const mockFunc = mockFunction.create(VerifyOrderConfig.create('onEnter', order));

    const keyCode = 44;
    const event = { keyCode };

    const props = {
      onEnter: mockFunc.getFunction(),
    };

    const component = mount(<Input {...props} />);
    component.find('input').simulate('onKeyDown'.substr(2).toLowerCase(), event);
    order.verify(arg => {});
  });

  it('props: null  Fired Enter', () => {
    const keyCode = 13;
    const event = { keyCode };
    const component = mount(<Input />);
    component.find('input').simulate('onKeyDown'.substr(2).toLowerCase(), event);
  });

  it('props: value changed Limited Input changed', () => {
    const value = '诸行无常';
    const changeValue = 'hello ligx';

    const component = mount(<LimitInputBox value={value} />);
    assertInputValue(component, value);
    component.find('input').simulate('change', { target: { value: changeValue } });
    assertInputValue(component, changeValue);
  });
  it('props: value changed Limited Input no changed', () => {
    const value = '诸行无常';
    const changeValue = 'hello ligx';

    const component = mount(<LimitInputBox value={value} />);
    assertInputValue(component, value);
    component.setProps({ value: changeValue });
    assertInputValue(component, value);
  });
  it('props: defaultValue', () => {
    const value = '诸行无常';
    const changeValue = 'hello ligx';

    class LimitInput extends React.Component<any, any> {
      render() {
        return <Input defaultValue={this.props.defaultValue} />;
      }
    }

    const component = mount(<LimitInput defaultValue={value} />);
    assertInputValue(component, value);
    component.find('input').simulate('change', { target: { value: changeValue } });
    assertInputValue(component, changeValue);
  });
  it('props: defaultValue & value', () => {
    const defaultValue = '诸行无常';
    const value = '诸法无我';
    const component = mount(<Input defaultValue={defaultValue} value={value} />);
    assertInputValue(component, value);
  });

  it('props: value changed Limited Input changed for prefix: Icon', () => {
    const value = '诸行无常';
    const changeValue = 'hello ligx';

    const component = mount(<LimitInputBox value={value} prefix={<div>hello</div>} />);
    const input = component.find('input').at(0);
    assertInputValue(component, value);
    input.find('input').simulate('change', { target: { value: changeValue } });
    assertInputValue(component, changeValue);
  });

  it('props: value changed Limited Input no changed  for prefix: Icon', () => {
    const value = '诸行无常';
    const changeValue = 'hello ligx';

    const component = mount(<LimitInputBox value={value} prefix={<div>hello</div>} />);
    assertInputValue(component, value);
    component.setProps({ value: changeValue });
    assertInputValue(component, value);
  });

  const parser = value => {
    return value.replace(/\$\s?|(,*)/g, '');
  };
  const formatter = value => {
    return `$${value}`.replace(/\B(?=(\d{3})+(?!\d))/g, ',');
  };
  const noop = value => {
    return value;
  };
  testOnChange('twice ["a", "b"] ', {}, ['a', 'b'], ['a', 'b'], ['a', 'b']);

  testOnChange(
    'props: onchange formatter&&parser both is null ',
    { formatter: noop, parser: noop },
    ['1234567'],
    ['1234567'],
    ['1234567']
  );
  testOnChange(
    'props: onchange parser:null ',
    { formatter, parser: noop },
    ['1234567'],
    ['$1,234,567'],
    ['1234567']
  );
  testOnChange(
    'props: onchange formatter:null ',
    { formatter: noop, parser },
    ['1234567'],
    ['1234567'],
    ['1234567']
  );
  testOnChange(
    'props: onchange formatter&&parser ',
    { formatter, parser },
    ['1234567'],
    ['$1,234,567'],
    ['1234567']
  );

  function testValidateTop(title: string, value: string, expValue: ValidateStatus) {
    it(` props :validateStatus  ${title}  &&  valiedateType top ;`, () => {
      const onChange = () => {};
      const component = mount(<TopInput validateType="top" onChange={onChange} />);
      component
        .find('input')
        .at(0)
        .simulate('change', { target: { value } });
      component
        .find('input')
        .at(0)
        .simulate('blur');
      expect(component.state().validateStatus).toBe(expValue);
    });
  }

  testValidateTop('value :""', '', 'default');
  testValidateTop('value :"111a1111"', '111a1111', 'error');
  testValidateTop('value :"a"', 'a', 'error');
  testValidateTop('value :"aaa"', 'aaa', 'error');

  function testValidateBottomAndInner(
    title: string,
    validateType: ValidateType,
    value: string,
    expValue: ValidateStatus
  ) {
    it(` props :validateStatus  ${title}  &&  valiedateType ${validateType} ;`, () => {
      const onChange = () => {};
      const component = mount(<ValidateInput validateType={validateType} onChange={onChange} />);
      const cmpInput = component.find('input').at(0);
      cmpInput.simulate('change', { target: { value } });
      expect(component.find(Widget.Input).props().validateStatus).toBe(expValue);
    });
  }

  testValidateBottomAndInner('value :""', 'bottom', '', 'default');
  testValidateBottomAndInner('value :"111"', 'bottom', '111', 'default');
  testValidateBottomAndInner('value :"111a1"', 'bottom', '111a1', 'error');
  testValidateBottomAndInner('value :"a"', 'bottom', 'a', 'error');
  testValidateBottomAndInner('value :"a"', 'bottom', 'aaa', 'error');

  it('props: readOnly true', () => {
    const value = '诸行无常';
    const changeValue = 'hello ligx';

    const component = mount(<Input value={value} readOnly={true} />);
    assertInputValue(component, value);
    component.find('input').simulate('change', { target: { value: changeValue } });
    assertInputValue(component, value);
  });
});
