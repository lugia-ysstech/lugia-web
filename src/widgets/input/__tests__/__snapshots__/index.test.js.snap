// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input props: getTheme: width:100,margin:10, 1`] = `
.c2 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  visibility: hidden;
  padding-right: 1rem;
}

.c1 {
  box-sizing: border-box;
  font-family: inherit;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  outline: none;
  height: 100%;
  width: 100%;
  border: none;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  cursor: text;
  border-top-left-radius: 0.4rem;
  border-top-right-radius: 0.4rem;
  border-bottom-right-radius: 0.4rem;
  border-bottom-left-radius: 0.4rem;
}

.c1::-webkit-input-placeholder {
  color: #cccccc;
  font-size: 1.4rem;
}

.c1::-moz-placeholder {
  color: #cccccc;
  font-size: 1.4rem;
}

.c1:-ms-input-placeholder {
  color: #cccccc;
  font-size: 1.4rem;
}

.c1::placeholder {
  color: #cccccc;
  font-size: 1.4rem;
}

.c0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  outline: none;
  min-width: 3rem;
  width: 100%;
  border-top-width: 0.1rem;
  border-top-style: solid;
  border-top-color: #cccccc;
  border-bottom-width: 0.1rem;
  border-bottom-style: solid;
  border-bottom-color: #cccccc;
  border-left-width: 0.1rem;
  border-left-style: solid;
  border-left-color: #cccccc;
  border-right-width: 0.1rem;
  border-right-style: solid;
  border-right-color: #cccccc;
  border-top-left-radius: 0.4rem;
  border-top-right-radius: 0.4rem;
  border-bottom-right-radius: 0.4rem;
  border-bottom-left-radius: 0.4rem;
}

<div
  className="c0"
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  style={
    Object {
      "height": "3.2rem",
      "marginBottom": "1rem",
      "marginLeft": "1rem",
      "marginRight": "1rem",
      "marginTop": "1rem",
      "width": "10rem",
    }
  }
>
  <input
    autoFocus={false}
    className="c1"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyPress={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    size="default"
    style={
      Object {
        "color": "#363B40",
        "fontSize": "1.4rem",
        "paddingBottom": "0rem",
        "paddingLeft": "1rem",
        "paddingRight": "1rem",
        "paddingTop": "0rem",
      }
    }
    type="text"
    value=""
  />
  <i
    className="lugia-icon-reminder_close   c2"
    disabled={false}
    onClick={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    style={
      Object {
        "color": "#9EA4AE",
        "fontSize": "1.2rem",
      }
    }
  />
</div>
`;

exports[`Input props: null 1`] = `
.c2 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  visibility: hidden;
  padding-right: 1rem;
}

.c1 {
  box-sizing: border-box;
  font-family: inherit;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  outline: none;
  height: 100%;
  width: 100%;
  border: none;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  cursor: text;
  border-top-left-radius: 0.4rem;
  border-top-right-radius: 0.4rem;
  border-bottom-right-radius: 0.4rem;
  border-bottom-left-radius: 0.4rem;
}

.c1::-webkit-input-placeholder {
  color: #cccccc;
  font-size: 1.4rem;
}

.c1::-moz-placeholder {
  color: #cccccc;
  font-size: 1.4rem;
}

.c1:-ms-input-placeholder {
  color: #cccccc;
  font-size: 1.4rem;
}

.c1::placeholder {
  color: #cccccc;
  font-size: 1.4rem;
}

.c0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  outline: none;
  min-width: 3rem;
  width: 100%;
  border-top-width: 0.1rem;
  border-top-style: solid;
  border-top-color: #cccccc;
  border-bottom-width: 0.1rem;
  border-bottom-style: solid;
  border-bottom-color: #cccccc;
  border-left-width: 0.1rem;
  border-left-style: solid;
  border-left-color: #cccccc;
  border-right-width: 0.1rem;
  border-right-style: solid;
  border-right-color: #cccccc;
  border-top-left-radius: 0.4rem;
  border-top-right-radius: 0.4rem;
  border-bottom-right-radius: 0.4rem;
  border-bottom-left-radius: 0.4rem;
}

<div
  className="c0"
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  style={
    Object {
      "height": "3.2rem",
    }
  }
>
  <input
    autoFocus={false}
    className="c1"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyPress={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    size="default"
    style={
      Object {
        "color": "#363B40",
        "fontSize": "1.4rem",
        "paddingBottom": "0rem",
        "paddingLeft": "1rem",
        "paddingRight": "1rem",
        "paddingTop": "0rem",
      }
    }
    type="text"
    value=""
  />
  <i
    className="lugia-icon-reminder_close   c2"
    disabled={false}
    onClick={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    style={
      Object {
        "color": "#9EA4AE",
        "fontSize": "1.2rem",
      }
    }
  />
</div>
`;

exports[`Input props: placeholder  1`] = `
.c2 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  visibility: hidden;
  padding-right: 1rem;
}

.c1 {
  box-sizing: border-box;
  font-family: inherit;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  outline: none;
  height: 100%;
  width: 100%;
  border: none;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  cursor: text;
  border-top-left-radius: 0.4rem;
  border-top-right-radius: 0.4rem;
  border-bottom-right-radius: 0.4rem;
  border-bottom-left-radius: 0.4rem;
}

.c1::-webkit-input-placeholder {
  color: #cccccc;
  font-size: 1.4rem;
}

.c1::-moz-placeholder {
  color: #cccccc;
  font-size: 1.4rem;
}

.c1:-ms-input-placeholder {
  color: #cccccc;
  font-size: 1.4rem;
}

.c1::placeholder {
  color: #cccccc;
  font-size: 1.4rem;
}

.c0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  outline: none;
  min-width: 3rem;
  width: 100%;
  border-top-width: 0.1rem;
  border-top-style: solid;
  border-top-color: #cccccc;
  border-bottom-width: 0.1rem;
  border-bottom-style: solid;
  border-bottom-color: #cccccc;
  border-left-width: 0.1rem;
  border-left-style: solid;
  border-left-color: #cccccc;
  border-right-width: 0.1rem;
  border-right-style: solid;
  border-right-color: #cccccc;
  border-top-left-radius: 0.4rem;
  border-top-right-radius: 0.4rem;
  border-bottom-right-radius: 0.4rem;
  border-bottom-left-radius: 0.4rem;
}

<div
  className="c0"
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  style={
    Object {
      "height": "3.2rem",
    }
  }
>
  <input
    autoFocus={false}
    className="c1"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyPress={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    placeholder="请填写内容"
    size="default"
    style={
      Object {
        "color": "#363B40",
        "fontSize": "1.4rem",
        "paddingBottom": "0rem",
        "paddingLeft": "1rem",
        "paddingRight": "1rem",
        "paddingTop": "0rem",
      }
    }
    type="text"
    value=""
  />
  <i
    className="lugia-icon-reminder_close   c2"
    disabled={false}
    onClick={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    style={
      Object {
        "color": "#9EA4AE",
        "fontSize": "1.2rem",
      }
    }
  />
</div>
`;

exports[`Input props: placeholder  2`] = `
.c2 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  visibility: hidden;
  padding-right: 1rem;
}

.c1 {
  box-sizing: border-box;
  font-family: inherit;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  outline: none;
  height: 100%;
  width: 100%;
  border: none;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  cursor: text;
  border-top-left-radius: 0.4rem;
  border-top-right-radius: 0.4rem;
  border-bottom-right-radius: 0.4rem;
  border-bottom-left-radius: 0.4rem;
}

.c1::-webkit-input-placeholder {
  color: #cccccc;
  font-size: 1.4rem;
}

.c1::-moz-placeholder {
  color: #cccccc;
  font-size: 1.4rem;
}

.c1:-ms-input-placeholder {
  color: #cccccc;
  font-size: 1.4rem;
}

.c1::placeholder {
  color: #cccccc;
  font-size: 1.4rem;
}

.c0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  outline: none;
  min-width: 3rem;
  width: 100%;
  border-top-width: 0.1rem;
  border-top-style: solid;
  border-top-color: #cccccc;
  border-bottom-width: 0.1rem;
  border-bottom-style: solid;
  border-bottom-color: #cccccc;
  border-left-width: 0.1rem;
  border-left-style: solid;
  border-left-color: #cccccc;
  border-right-width: 0.1rem;
  border-right-style: solid;
  border-right-color: #cccccc;
  border-top-left-radius: 0.4rem;
  border-top-right-radius: 0.4rem;
  border-bottom-right-radius: 0.4rem;
  border-bottom-left-radius: 0.4rem;
}

<div
  className="c0"
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  style={
    Object {
      "height": "3.2rem",
    }
  }
>
  <input
    autoFocus={false}
    className="c1"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyPress={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    placeholder=""
    size="default"
    style={
      Object {
        "color": "#363B40",
        "fontSize": "1.4rem",
        "paddingBottom": "0rem",
        "paddingLeft": "1rem",
        "paddingRight": "1rem",
        "paddingTop": "0rem",
      }
    }
    type="text"
    value=""
  />
  <i
    className="lugia-icon-reminder_close   c2"
    disabled={false}
    onClick={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    style={
      Object {
        "color": "#9EA4AE",
        "fontSize": "1.2rem",
      }
    }
  />
</div>
`;

exports[`Input props: size 1`] = `
.c2 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  visibility: hidden;
  padding-right: 1rem;
}

.c1 {
  box-sizing: border-box;
  font-family: inherit;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  outline: none;
  height: 100%;
  width: 100%;
  border: none;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  cursor: text;
  border-top-left-radius: 0.4rem;
  border-top-right-radius: 0.4rem;
  border-bottom-right-radius: 0.4rem;
  border-bottom-left-radius: 0.4rem;
}

.c1::-webkit-input-placeholder {
  color: #cccccc;
  font-size: 1.2rem;
}

.c1::-moz-placeholder {
  color: #cccccc;
  font-size: 1.2rem;
}

.c1:-ms-input-placeholder {
  color: #cccccc;
  font-size: 1.2rem;
}

.c1::placeholder {
  color: #cccccc;
  font-size: 1.2rem;
}

.c0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  outline: none;
  min-width: 3rem;
  width: 100%;
  border-top-width: 0.1rem;
  border-top-style: solid;
  border-top-color: #cccccc;
  border-bottom-width: 0.1rem;
  border-bottom-style: solid;
  border-bottom-color: #cccccc;
  border-left-width: 0.1rem;
  border-left-style: solid;
  border-left-color: #cccccc;
  border-right-width: 0.1rem;
  border-right-style: solid;
  border-right-color: #cccccc;
  border-top-left-radius: 0.4rem;
  border-top-right-radius: 0.4rem;
  border-bottom-right-radius: 0.4rem;
  border-bottom-left-radius: 0.4rem;
}

<div
  className="c0"
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  style={
    Object {
      "height": "2.4rem",
    }
  }
>
  <input
    autoFocus={false}
    className="c1"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyPress={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    size="small"
    style={
      Object {
        "color": "#363B40",
        "fontSize": "1.2rem",
        "paddingBottom": "0rem",
        "paddingLeft": "1rem",
        "paddingRight": "1rem",
        "paddingTop": "0rem",
      }
    }
    type="text"
    value=""
  />
  <i
    className="lugia-icon-reminder_close   c2"
    disabled={false}
    onClick={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    style={
      Object {
        "color": "#9EA4AE",
        "fontSize": "1rem",
      }
    }
  />
</div>
`;

exports[`Input props: size 2`] = `
.c2 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  visibility: hidden;
  padding-right: 1rem;
}

.c1 {
  box-sizing: border-box;
  font-family: inherit;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  outline: none;
  height: 100%;
  width: 100%;
  border: none;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  cursor: text;
  border-top-left-radius: 0.4rem;
  border-top-right-radius: 0.4rem;
  border-bottom-right-radius: 0.4rem;
  border-bottom-left-radius: 0.4rem;
}

.c1::-webkit-input-placeholder {
  color: #cccccc;
  font-size: 1.4rem;
}

.c1::-moz-placeholder {
  color: #cccccc;
  font-size: 1.4rem;
}

.c1:-ms-input-placeholder {
  color: #cccccc;
  font-size: 1.4rem;
}

.c1::placeholder {
  color: #cccccc;
  font-size: 1.4rem;
}

.c0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  outline: none;
  min-width: 3rem;
  width: 100%;
  border-top-width: 0.1rem;
  border-top-style: solid;
  border-top-color: #cccccc;
  border-bottom-width: 0.1rem;
  border-bottom-style: solid;
  border-bottom-color: #cccccc;
  border-left-width: 0.1rem;
  border-left-style: solid;
  border-left-color: #cccccc;
  border-right-width: 0.1rem;
  border-right-style: solid;
  border-right-color: #cccccc;
  border-top-left-radius: 0.4rem;
  border-top-right-radius: 0.4rem;
  border-bottom-right-radius: 0.4rem;
  border-bottom-left-radius: 0.4rem;
}

<div
  className="c0"
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  style={
    Object {
      "height": "4rem",
    }
  }
>
  <input
    autoFocus={false}
    className="c1"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyPress={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    size="large"
    style={
      Object {
        "color": "#363B40",
        "fontSize": "1.4rem",
        "paddingBottom": "0rem",
        "paddingLeft": "1rem",
        "paddingRight": "1rem",
        "paddingTop": "0rem",
      }
    }
    type="text"
    value=""
  />
  <i
    className="lugia-icon-reminder_close   c2"
    disabled={false}
    onClick={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    style={
      Object {
        "color": "#9EA4AE",
        "fontSize": "1.2rem",
      }
    }
  />
</div>
`;
